### ISSUE TITLE
Underlying Customer Availability

### ISSUE THREAD

**Issue by msundvick** [kynac1/project47_last_mile_logistics#9]
So we have time windows, and a way to set the probability that a customer is futile from the multiday simulation (I think).
What we want is to have a true customer availability, and separate that from the time windows we use. We've talked about this before, but I thought I'd write down some actual ideas on how to achieve this.

True customer availability is not random. I like to think of it as a function that takes the current time and returns true or false.

We don't ever know this properly. To add some randomness, we can instead have a function that converts a time to a probability that the customer is present.

How we define these functions doesn't really matter, but we still need to get some time windows. Ideally, we'd select a time window that maximises the probability that the customer is available within the time window. If we have either of the above functions, this is easy for a given number of time windows and time window sizes.

The question is what information does the customer provide us. I think we start with some function for the true state, as a function of location and distance. This can be used in the simulation, but isn't known by the routing solver. We fit a model to this, which gives us the probabilistic state, and we assume that this is known by the routing solver. We could somehow use this directly, or create a set of time windows that works with ortools.

Now, lets say the customer provides us with some time windows (In practice we'll probably just randomly generate these from a limited set). This situation sounds like a prior/posterior thing, like a small bayesian network? We can update the probabilistic state function we already know with the new information, and do the same optimisation steps.

If the time windows change during travel, this can just be resolved as before, I think. The tricky part is if we phone ahead. Phoning ahead is kinda a function that returns true if they pick up and will be there, false if they pick up and tell us they won't, or a null result where they tell us nothing. This seems tricky to reason about, in terms of what is the likelyhood of each of these events occurring, but if we can get this then it should be simple to reschedule accordingly.

This is longer than I intended, but I thought I'd better write this stuff down before I forget it. 

Basically the short version is we need some true function for underlying availability. Until we have proper data, I guess we just make up some random probabilistic model for it.
